twoliter/embedded/Makefile.toml

```diff
@@ -354,11 +354,25 @@ dependencies = ["setup"]
 script_runner = "bash"
 script = [
 '''
+declare -a CARGO_MANIFESTS
+declare -a CARGO_DIGESTS
 for ws in sources variants .; do
   [ -s "${BUILDSYS_ROOT_DIR}/${ws}/Cargo.toml" ] || continue
-  cargo fetch --locked --manifest-path "${BUILDSYS_ROOT_DIR}/${ws}/Cargo.toml"
+  CARGO_MANIFESTS+=(--sync "${BUILDSYS_ROOT_DIR}/${ws}/Cargo.toml")
+  CARGO_DIGESTS+=("$(sha256sum "${BUILDSYS_ROOT_DIR}/${ws}/Cargo.lock" | awk '{ print $1 }')")
 done
-
+DIGEST="$(echo ${CARGO_DIGESTS[@]} | sha256sum | awk '{ print $1 }')"
+MARKER_FILE=".cargo/vendor-${DIGEST}.marker"
+if [ -f "${MARKER_FILE}" ]; then
+    echo "no changes detected in cargo projects, skipping cargo vendor"
+else
+    cargo vendor --locked ${CARGO_MANIFESTS[@]} .cargo/vendor > .cargo/config.toml 2> .cargo/vendor.log || {
+        echo "vendoring cargo sources failed"
+        cat .cargo/vendor.log
+    }
```

Comment:
We don't pass `--no-delete` to `cargo vendor` but we also don't clean up older marker files. I think we have to do one or the other.

In order to keep old marker files, I think we have to pass `--versioned-dirs` to `cargo vendor` so that we can keep multiple different versions of vendored deps based on the Bottlerocket version. But I'm not sure! Any chance you could test by doing `git checkouts` on workspaces with different versions of deps in `Cargo.lock` and see what happens?

Additionally, can we pass `--respect-source-config` here? I'm somewhat interested in using kits to ship cargo registries so that they can ship library dependencies. This idea isn't one that I've scoped or has much buy-in yet, but I'd like to keep it available as an option if we can.